src/Features/LanguageServer/ProtocolUnitTests/HandlerTests.cs

```diff
@@ -179,17 +180,15 @@ public Task<string> HandleRequestAsync(RequestContext context, CancellationToken
             }
         }
 
-        [ExportCSharpVisualBasicStatelessLspService(typeof(NotificationHandler)), PartNotDiscoverable, Shared]
         [LanguageServerEndpoint(MethodName, LanguageServerConstants.DefaultLanguageName)]
         internal class NotificationHandler : ILspServiceNotificationHandler<TestRequestTypeOne>
         {
             public const string MethodName = nameof(NotificationHandler);
-            public static readonly TaskCompletionSource<string> ResultSource = new();
+            public readonly TaskCompletionSource<string> ResultSource;
 
-            [ImportingConstructor]
-            [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
             public NotificationHandler()
             {
+                ResultSource = new TaskCompletionSource<string>();
```

Comment:
nit.  move to field initializer.
nit: move to primary constructor.

Reply:
Wasn't able to do this for some reason:
![image](https://github.com/dotnet/roslyn/assets/5749229/2d05722e-37eb-47af-858b-b5d30847458a)

Maybe because these tests target net472?

Reply:
nevermind, just dumb, fixing.